lib/DocumentService.php

```diff
@@ -142,17 +157,62 @@ public function getDocumentByUserId(string $userId, int $fileId, ?string $dir) :
 				return $this->reportError('Document for the fileId ' . $fileId . 'not found');
 			}
 
-			// Set basic parameters
+			/** @var \OCP\Files\Storage\IStorage $storage */
+			$storage = $document->getStorage();
+			$isSharedFile = $storage->instanceOfStorage('\OCA\Files_Sharing\SharedStorage');
+			$isFederatedShare = $storage->instanceOfStorage('\OCA\Files_Sharing\External\Storage');
```

Comment:
Any other alternative? Checking for specific classes doesn't look good

Reply:
- it is public method of that class, so why not to use it?
- I had extensive look and this is the only way in OC currently, probably we would need to somehow handle this better in OC10 core?

Reply:
There could be other classes implementing a shared storage. I know there are other places where this kind of check is done, so maybe it's too late and switching implementations is no longer an option.

Anyway, if there is no better option, we'll have to go with this at least for now.